x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/enrollment/CreateEnrollmentTokenTests.java

```diff
@@ -0,0 +1,238 @@
+                new SecureString("elastic")));
+            assertThat(ex.getMessage(), Matchers.containsString("Unexpected response code [400] from calling POST "));
+        } catch (Exception e) {
+            logger.info("testFailedCreateApiKey failed", e);
+            fail("testFailedCreateApiKey failed");
+        }
+    }
+
+    public void testFailedRetrieveHttpInfo() {
+        try {
+            final CommandLineHttpClient client = mock(CommandLineHttpClient.class);
+            when(client.getDefaultURL()).thenReturn("http://localhost:9200");
+            final URL defaultUrl = new URL(client.getDefaultURL());
+            final URL createAPIKeyURL = CreateEnrollmentToken.createAPIKeyURL(defaultUrl);
+            final URL getHttpInfoURL = CreateEnrollmentToken.getHttpInfoURL(defaultUrl);
+
+            final HttpResponse httpResponseOK = new HttpResponse(HttpURLConnection.HTTP_OK, new HashMap<String, Object>());
+            when(client.execute(anyString(), any(URL.class), anyString(), any(SecureString.class), any(CheckedSupplier.class),
+                any(CheckedFunction.class))).thenReturn(httpResponseOK);
+
+            String createApiKeyResponseBody;
+            try (XContentBuilder builder = XContentFactory.contentBuilder(XContentType.JSON)) {
+                builder.startObject()
+                    .field("id", "DR6CzXkBDf8amV_48yYX")
+                    .field("name", "enrollment_token_API_key_VuaCfGcBCdbkQm")
+                    .field("expiration", "1622652381786")
+                    .field("api_key", "x3YqU_rqQwm-ESrkExcnOg")
+                    .endObject();
+                createApiKeyResponseBody = Strings.toString(builder);
+            }
+            when(client.execute(eq("POST"), eq(createAPIKeyURL), eq(ElasticUser.NAME), any(SecureString.class),
+                any(CheckedSupplier.class), any(CheckedFunction.class)))
+                .thenReturn(createHttpResponse(HttpURLConnection.HTTP_OK, createApiKeyResponseBody));
+
+            final HttpResponse httpResponseNotOK = new HttpResponse(HttpURLConnection.HTTP_BAD_REQUEST, new HashMap<String, Object>());
+            when(client.execute(anyString(), eq(getHttpInfoURL), anyString(), any(SecureString.class), any(CheckedSupplier.class),
+                any(CheckedFunction.class))).thenReturn(httpResponseNotOK);
+
+            final CreateEnrollmentToken createEnrollmentToken = new CreateEnrollmentToken(environment, client);
+            UserException ex = expectThrows(UserException.class, () -> createEnrollmentToken.create(terminal, "elastic",
+                new SecureString("elastic")));
+            assertThat(ex.getMessage(), Matchers.containsString("Unexpected response code [400] from calling GET "));
+        } catch (Exception e) {
+            logger.info("testFailedRetrieveHttpInfo failed", e);
+            fail("testFailedRetrieveHttpInfo failed");
+        }
+    }
+
+    public void testSSLNotConfigured() {
```

Comment:
Isn't this actually testing SSLService ? Don't we have this test already in SSLServiceTests ? If not, maybe that's a better place to put such a test?